### DISCUSSION TITLE
When using the language switcher and filament-translatable plugin , relationships don't get switched or updated

### DISCUSSION THREAD

**Discussion by nabileisa** [filamentphp/filament#9549] [Help] [answered]
Hello,
I'm attempting to build a multi lingual app using laravel and filament, 
and I have noticed a few odd behaviors  that I wish you would clarify for me.
I added the translatable plugin following the docs

- In the table list when using the language switcher only direct properties of the model get updated to selected language example: `Tables\Columns\TextColumn::make('name')`
any other propreties of relationship models loaded using the dot notation  
`Tables\Columns\TextColumn::make('owner.name')`
 remain the same original language. 
- this is affecting the tables in the list page as well as the `RelatiohshipManager` tables too.
- Same thing happen when using relationship select, values don't get updated when language switcher change.
- I have managed to overcome some of the other limitations that I have encountered with forms and the translation plugin by taking out the `LocaleSwitcher` and `Translatable` trait completely from form pages and separating each language in it's own tab.

But I have reached a road block here with tables and relationships.
So can you help me understand why it won't work and how I can fix it .
thank you

**Comment by nabileisa** [comment, marked as answer]
Never mind , I found a way to make it work by using a closure and injecting `$livewire` and getting active locale from it `$livewire->activeLocale` and loading translated the relationship data using it.

**Reply by christopheraseidl** [reply]
Could you share your solution?
